libs/algorithms/include/hpx/parallel/algorithms/transform.hpp

```diff
@@ -893,47 +1136,95 @@ namespace hpx { namespace parallel { inline namespace v1 {
     }
 }}}    // namespace hpx::parallel::v1
 
-#if defined(HPX_HAVE_THREAD_DESCRIPTION)
-namespace hpx { namespace traits {
-    template <typename ExPolicy, typename F, typename Proj>
-    struct get_function_address<
-        parallel::v1::detail::transform_iteration<ExPolicy, F, Proj>>
+namespace hpx {
+
+    ///////////////////////////////////////////////////////////////////////////
+    // CPO for hpx::transform
+    HPX_INLINE_CONSTEXPR_VARIABLE struct transform_t final
+      : hpx::functional::tag<transform_t>
     {
-        static std::size_t call(
-            parallel::v1::detail::transform_iteration<ExPolicy, F, Proj> const&
-                f) noexcept
+    private:
+        // clang-format off
+        template <typename FwdIter1, typename FwdIter2,
+            typename F, HPX_CONCEPT_REQUIRES_(
+                hpx::traits::is_iterator<FwdIter1>::value &&
+                hpx::traits::is_iterator<FwdIter2>::value
+            )>
+        // clang-format on
+        friend FwdIter2 tag_invoke(hpx::transform_t, FwdIter1 first,
+            FwdIter1 last, FwdIter2 dest, F&& f)
         {
-            return get_function_address<
-                typename hpx::util::decay<F>::type>::call(f.f_);
+            return std::transform(first, last, dest);
```

Comment:
Calling the standard algorithms will have the correct semantics, but exposes inconsistent exception behavior. For the other algorithms we have settled on relying our own implementations even in those cases where semantically the standard version would be sufficient.